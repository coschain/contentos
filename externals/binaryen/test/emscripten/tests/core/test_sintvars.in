#include <stdio.h>
struct S {
  char *match_start;
  char *strstart;
};
int main() {
  struct S _s;
  struct S *s = &_s;
  unsigned short int sh;

  s->match_start = (char *)32522;
  s->strstart = (char *)(32780);
  printf("*%d,%d,%d*\n", (int)s->strstart, (int)s->match_start,
         (int)(s->strstart - s->match_start));
  sh = s->strstart - s->match_start;
  printf("*%d,%d*\n", sh, sh >> 7);

  s->match_start = (char *)32999;
  s->strstart = (char *)(32780);
  printf("*%d,%d,%d*\n", (int)s->strstart, (int)s->match_start,
         (int)(s->strstart - s->match_start));
  sh = s->strstart - s->match_start;
  printf("*%d,%d*\n", sh, sh >> 7);
}
