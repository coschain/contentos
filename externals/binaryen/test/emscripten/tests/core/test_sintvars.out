*32780,32522,258*
*258,2*
*32780,32999,-219*
*65317,510*